src/UI/Shapes.java

```diff
@@ -0,0 +1,257 @@
+package UI;
+
+/**
+ * @author: Sajith Kumar Erasani
+ * This controller handles all the action and mouse events of canvas for drawing and selecting Shapes
+ */
+
+import processing.utility.*;
+import processing.*;
+import infrastructure.validation.logger.*;
+import javafx.scene.canvas.Canvas;
+import javafx.scene.canvas.GraphicsContext;
+import javafx.scene.paint.Color;
+
+public class Shapes {
+
+	static boolean rectselected = false;
+	static boolean circleselected = false;
+	static boolean lineselected = false;
+	static boolean triangleselected = false;
+	static boolean squareselected = false;
+	static boolean eraserselected = false;
+	static IDrawShapes drawshape = ProcessingFactory.getProcessor();
+	static ILogger logger = LoggerFactory.getLoggerInstance();
+
+	/**
+	 * This method will update the shape selection to be default
+	 */
+	public static void defaultSelected() {
+		rectselected = false;
+		circleselected = false;
+		lineselected = false;
+		triangleselected = false;
+		squareselected = false;
+		eraserselected = false;
+	}
+
+    /**
+     * This method will draw rectangle and sends the data to processing module
+     * @param color: Color selected from color picker
+     * @param g: object of graphicscontext
+     */
+    public static void drawPerfectRect(Color color,GraphicsContext g, double x12, double y12, double d, double e) {
+        double px = Math.min(x12,d);
+        double py = Math.min(y12,e);
+        double pw=Math.abs(x12-d);
+        double ph=Math.abs(y12-e);
+        g.setStroke(color);
+        g.strokeRect(px, py, pw, ph);
+        Position  start = new Position((int) (px*100),(int) (py*100));
+        Position end = new Position((int) (px+pw)*100,(int) (py+ph)*100);
+        Intensity i = new Intensity((int) color.getRed(),(int) color.getGreen(),(int) color.getBlue());
+        Pixel p1 = new Pixel(start,i);
+        Pixel p2 = new Pixel(end,i);
+        drawshape.drawRectangle(p1,p2);
+        logger.log(ModuleID.UI, LogLevel.INFO, "Rectangle drawn");
```

Comment:
```suggestion
        logger.log(
            ModuleID.UI, 
            LogLevel.INFO, 
            "Rectangle drawn"
         );
```

Reply:
In the log message, coordinates of the rectangle can be included. It will help us to distinguish between two quick rectangles.

Reply:
Added coordinates for all shapes in log messages